### ISSUE TITLE
Build fails with rust 1.80

### ISSUE THREAD

**Issue by GaetanLepage** [sourcegraph/sg.nvim#259]
When trying to build the plugin with rust >= 1.80, I get:

```
error[E0282]: type annotations needed for `Box<_>`
  --> /home/gaetan/.local/share/cargo/registry/src/index.crates.io-6f17d22bba15001f/time-0.3.30/src/format_description/parse/mod.rs:83:9
   |
83 |     let items = format_items
   |         ^^^^^
...
86 |     Ok(items.into())
   |              ---- type must be known at this point
   |
help: consider giving `items` an explicit type, where the placeholders `_` are specified
   |
83 |     let items: Box<_> = format_items
   |              ++++++++
```

This should be fixed by a simple dependency bump (see https://github.com/NixOS/nixpkgs/issues/332957).
However, running `cargo update` leads to a new error:
```
error[E0283]: type annotations needed
   --> /home/gaetan/.local/share/cargo/registry/src/index.crates.io-6f17d22bba15001f/gix-url-0.22.0/src/lib.rs:106:54
    |
106 |             let path = gix_path::from_bstr(self.path.as_ref());
    |                                                      ^^^^^^
    |
    = note: multiple `impl`s satisfying `BString: AsRef<_>` found in the `bstr` crate:
            - impl AsRef<BStr> for BString;
            - impl AsRef<[u8]> for BString;
help: try using a fully qualified path to specify the expected types
    |
106 |             let path = gix_path::from_bstr(<BString as AsRef<T>>::as_ref(&self.path));
    |                                            +++++++++++++++++++++++++++++++         ~

error[E0283]: type annotations needed
   --> /home/gaetan/.local/share/cargo/registry/src/index.crates.io-6f17d22bba15001f/gix-url-0.22.0/src/lib.rs:106:24
    |
106 |             let path = gix_path::from_bstr(self.path.as_ref());
    |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ cannot infer type for reference `&_`
    |
    = note: multiple `impl`s satisfying `Cow<'_, BStr>: From<&_>` found in the `bstr` crate:
            - impl<'a> From<&'a BStr> for Cow<'a, BStr>;
            - impl<'a> From<&'a BString> for Cow<'a, BStr>;
    = note: required for `&_` to implement `Into<Cow<'_, BStr>>`
note: required by a bound in `from_bstr`
   --> /home/gaetan/.local/share/cargo/registry/src/index.crates.io-6f17d22bba15001f/gix-path-0.9.0/src/convert.rs:126:34
    |
126 | pub fn from_bstr<'a>(input: impl Into<Cow<'a, BStr>>) -> Cow<'a, Path> {
    |                                  ^^^^^^^^^^^^^^^^^^^ required by this bound in `from_bstr`

For more information about this error, try `rustc --explain E0283`.
error: could not compile `gix-url` (lib) due to 2 previous errors
```

**Comment by supermarin** [comment]
I fixed it in #262 but not sure if this is maintained anymore since TJ left.
@burmudar @jdorfman any chance we can have some :eyes: on this?

**Comment by burmudar** [comment]
> I fixed it in #262 but not sure if this is actively maintained anymore. @burmudar @jdorfman any chance we can have some 👀 on this?

Awesome! The workflow needed an approval to run so just kicked that off!

**Comment by burmudar** [comment]
Thanks for reporting @GaetanLepage and thanks for fixing it @supermarin. I've merged it now 🥳
